recipes-bsp/libimxdmabuffer/libimxdmabuffer_1.1.2.bb

```diff
@@ -34,6 +28,17 @@ EXTRA_OECONF = "--imx-linux-headers-path=${STAGING_INCDIR_IMX} \
 LIBG2D_PACKAGECONFIG = "g2d"
 LIBG2D_PACKAGECONFIG:imxdpu = ""
 
+# Device node path to the DMA-BUF heap to use in the dma-heap allocator. This
+# must be one that allocates physically contiguous memory, otherwise it cannot
+# allocate usable DMA buffers. An uncached DMA heap is selected to avoid issues
+# with cache coherence. Note that the uncached heap is not available prior to
+# imx-kernel 5.15.5; older kernels should use "/dev/dma_heap/linux,cma" instead
+# and remove the "--dma-heap-uncached-memory" switch below from DMA_HEAP_CONF.
+DMA_HEAP_DEVICE_NODE_PATH = "/dev/dma_heap/linux,cma-uncached"
+
+DMA_HEAP_CONF = "--dma-heap-device-node-path=${DMA_HEAP_DEVICE_NODE_PATH} \
+                 --dma-heap-uncached-memory"
+
```

Comment:
I think what Otavio means below is to define two package configs `dma-heap-cached` and `dma-heap-uncached`. Then the user can more easily select the cached heap.

Reply:
Ah, true, right now it cannot be selected just by setting one value. Although, is `PACKAGECONFIG` really the best option for this? What about a separate `USE_UNCACHED_DMA_HEAP` variable that can be set to 1 or 0?

Reply:
I prefer the `PACKAGECONFIG` approach here because it elevates that choice into the `PACKAGECONFIG` list itself.

> I mean, you wouldn't be able to add both dma-heap-cached and dma-heap-uncached to PACKAGECONFIG.

You can set the sixth field of the `PACKAGECONFIG` variable to capture this mutual exclusion.

Reply:
Alright, I will use `PACKAGECONFIG` then.